src/main/java/com/dsec/backend/service/UserService.java

```diff
@@ -126,22 +135,98 @@ public UserInfoDTO login(LoginInfoDTO dto, HttpServletResponse response) {
 	}
 
 	@Override
-	public UserInfoDTO getUser(Jwt user) {
+	public ResponseEntity<?> getAllUsers() throws JsonProcessingException {
+		final ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+		int i = 0;
+		String responseBody = new String();
+		for (UserModel userModel : userRepository.findAll()) {
+			String tmpString = "";
+			if(i != 0)
+				tmpString = ",";
+			tmpString = tmpString + ("\""+i+"\":"+mapper.writeValueAsString(new UserInfoDTO(userModel.getId(), userModel.getEmail(), userModel.getFirstName(), userModel.getLastName(), userModel.getUserRole())));
+			responseBody = responseBody + tmpString;
+			i++;
+		}
 
-		try {
-			UserPrincipal userPrincipal =
-					objectMapper.readValue((String) user.getClaim("object"), UserPrincipal.class);
+		if(responseBody.isEmpty())
+			return ResponseEntity.noContent().build();
 
-			UserModel userModel = userPrincipal.getUserModel();
+		return ResponseEntity.ok( responseBody);
+	}
 
-			return new UserInfoDTO(userModel.getId(), userModel.getEmail(),
-					userModel.getFirstName(), userModel.getLastName(), userModel.getUserRole());
+	@Override
+	public ResponseEntity<?> getUser(int id)
+	{
+		try
+		{
+			UserModel userModel = userRepository.findById(id);
+			Jwt user = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+			JSONObject userData = new JSONObject((String) user.getClaim("object"));
+			final ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+			if(userData.getJSONObject("userModel").getInt("id") == userModel.getId())
+				return ResponseEntity.ok(mapper.writeValueAsString(new UserInfoDTO(userModel.getId(), userModel.getEmail(), userModel.getFirstName(), userModel.getLastName(), userModel.getUserRole())));
+
+			return ResponseEntity.ok(mapper.writeValueAsString(new UserInfoRestricedDTO(userModel.getId(),userModel.getEmail(), userModel.getFirstName(), userModel.getLastName())));
+
+		} catch(Exception e) {
+			return ResponseEntity.notFound().build();
+		}
+	}
 
-		} catch (Exception e) {
-			logger.error("UserPrincipal deserialization error {}", e.getMessage());
+	@Override
+	public ResponseEntity<?> editUser(int id, UserDTO userDTO)
+	{
+		try
+		{
+			if(id == userDTO.id())
+			{
+				UserModel userModel = userRepository.findById(id);
+				Jwt user = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
There is a better way to obtain this using the @AuthenticationPrincipal annotation. Please refer to [this](https://www.springcloud.io/post/2022-02/spring-security-get-current-user/#authenticationprincipal).